### ISSUE TITLE
iOS `Numeric Format` setting not respected

### ISSUE THREAD

**Issue by LiamMorrow** [dotnet/runtime#110430]
### Description

When using `decimal.Parse` using the default `CultureInfo` (or not specifying one), MAUI does not respect the user's decimal separator options in iOS.  Instead, it simply uses the language's default.  What this means is that when a user is prompted for numerical, decimal input, the iOS keyboard presents them with a `comma` separator (as they have configured), but decimal.Parse treats it as a grouping separator.

Note that if the user specifies a LANGUAGE which uses commas as a decimal separator, everything works as expected and decimal parse treats commas as decimal separators.

<details>
  <summary> Screenshot showing how the default culture parses a decimal, and the correct parsing</summary>

 ![Image](https://github.com/user-attachments/assets/c151e654-2a35-465c-b874-42e4d334f294)

</details>

<details><summary>Screenshot showing that the keyboard only allows entering decimals with commas</summary>

![Image](https://github.com/user-attachments/assets/d458cc74-4121-4f29-b4b7-ebd35193a22a)

</details>

<details>
<summary> Code driving the parsing logic</summary>

```
 if (decimal.TryParse(e.NewTextValue, out decimal result))
        {
            resultLabel.Text = $"Parsed value with default culture: {result}";
        }
        else
        {
            resultLabel.Text = "Invalid input";
        }
#if IOS
        var correctedCulture = CultureInfo.CurrentCulture.Clone() as CultureInfo;
        correctedCulture.NumberFormat.NumberDecimalSeparator = NSLocale
            .CurrentLocale
            .DecimalSeparator;
        correctedCulture.NumberFormat.NumberGroupSeparator = NSLocale
            .CurrentLocale
            .GroupingSeparator;

        if (decimal.TryParse(e.NewTextValue, correctedCulture, out decimal result2))
        {
            resultLabel2.Text = $"Parsed value with culture from IOS settings: {result2}";
        }
        else
        {
            resultLabel2.Text = "Invalid input";
        }
#endif
```

</details>

<details><summary>Video showing off behaviour</summary>

https://github.com/user-attachments/assets/1b488061-38eb-4e40-9704-ebc8dc06f89a

</details>

### Steps to Reproduce

1. Open iOS settings -> General -> Language
2. Set language to English
3. Set Number Format to 1.234.234,83 (the one which uses comma as a decimal separator)
4. In a Maui app type in a numeric editor (notice iOS presents only a comma as the decimal separator)
5. Notice that decimal.Parse treats the comma as a grouping separator

An example repo for parsing can be found here:
https://github.com/LiamMorrow/maui-decimal-repro

### Link to public reproduction project repository

https://github.com/LiamMorrow/maui-decimal-repro

### Version with bug

9.0.12 SR1.2

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 18

### Did you find any workaround?

Cloning the current culture, then using the proper separators in the clone, and setting it as the current culture on app startup does fix this.  See [this PR](https://github.com/LiamMorrow/LiftLog/pull/335) for an example.

### Relevant log output

```shell

```

**Comment by tarekgh** [comment]
@matouskozak could you please have a look and triage?

**Comment by LiamMorrow** [comment]
For the record, this was originally reported here:
https://github.com/LiamMorrow/LiftLog/issues/332

Note that the user seems to think it is due to the mix of Swedish&English keyboard

**Comment by matouskozak** [comment]
@LiamMorrow Thanks a lot for the great issue description with concrete examples. I'm able to reproduce the issue, on both .net 8 and .net 9 MAUI apps.

Do you know when was the last time this used to work correctly?

Anyways, I'll investigate why the `CurrentCulture` is not set properly with the iOS number format setting.

**Comment by LiamMorrow** [comment]
Thanks @matouskozak for looking into it!

> Do you know when was the last time this used to work correctly?

I have a feeling it has never worked correctly, however iOS 18 has released new keyboards which allow for multi locale keyboards which has probably made it much easier for it to happen.  I don't think many users would stray from their default locale's number format otherwise.  The user who reported it to me says it started occuring when they switched to the Swedish/English keyboard.

**Comment by jmacmullin** [comment]
Any update on this?

**Comment by jmacmullin** [comment]
@matouskozak where are you getting the value that you use for the string locale identifier? And has this changed recently?

I was directed at this issue by @jfversluis who closed an issue I opened in the MAUI project where we're observing a change in behaviour between the previous version of MAUI/.net and the latest:
https://github.com/dotnet/maui/issues/27793

In that issue I describe (and include a sample project that reproduces the issue) the fact that when running on a device's with the region set to Australia we'd expect MAUI/.net to respect the device's locale and use Australian date formats (dd/mm/yyyy) - and this is what used to happen prior to .net9 and the latest version of MAUI. Now it looks like `System.Globalization.CultureInfo.CurrentCulture` doesn't reflect the device's settings on iOS - though it does seem to in a console app running on macOS.

Is this the same underlying issue?

**Comment by matouskozak** [comment]
@tarekgh I took a look again at this issue and it appears that on Unix systems we do not respect user overrides and use the default culture setting. Is this an intentional decision? I could add a dedicated apple mobile code to handle this but that would create an inconsistency with macOS. (Note: on Android we don't hit this issue as it doesn't allow to change the decimal separator).

**Comment by tarekgh** [comment]
Right, this isn't really a general "Unix" decision, it's specific to how each OS exposes locale data.

On Linux (Ubuntu and friends) there is no per-field user override to respect. The locale is selected as a whole through the POSIX environment variables (`LC_ALL`, `LC_NUMERIC`, `LANG`, ...). Even the GNOME "Region & Language -> Formats" option just points those variables at a different complete locale, it does not let a user override an individual field like the decimal separator while keeping the language. ICU then resolves formatting from CLDR for that locale, so there is nothing extra to pick up.

Apple is different. On iOS/macOS the OS does have real per-field overrides (the "Number Format" setting here maps to `NSLocale.DecimalSeparator`, `NSLocale.GroupingSeparator`, etc.), and those are independent of the chosen language. Our ICU based path just doesn't consult them today, which is what this bug is about.

So adding Apple specific handling would not create an inconsistency with Linux, because Linux has no such setting. It would need to be done for macOS as well so iOS and macOS stay aligned. macOS has the same class of problem tracked in https://github.com/dotnet/runtime/issues/61148, so we should treat these together rather than special casing iOS on its own.
